### ISSUE TITLE
[SCREENREADER]: Alert icons MUST be implemented in a meaningful way for assistive technology

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#8344]
## Scope for this issue: 

- Limited to alert icons displayed in search results cards
- We will request separate feedback on facility detail pages alerts rather than applying this feedback to those areas. 

# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The icons in the alerts **must** be implemented in a way that be used or ignored by assistive technology. In this case, these icons mean "alert" to a sighted user, so they should convey that for non-sighted users, as well. The additional context that the operating status alert is temporary and due to COVID-19 **_(please see Michelle's addendum below)_** should be included, because hearing a search result and "facility closed" can leave a Veteran asking, **"Then WHY is this in my search results?!**

I marked this as 508-defect-2 because the missing data would be crucial for understanding.

This change applies to both the alert "lozenge" in the search result and the alert on the operating status page. Both instances would aid non-sighted users if screen reader only text indicated "Alert" before reading "Facility Closed" or other message

I uploaded to Slack [recordings of VoiceOver, NVDA, and JAWS](https://dsva.slack.com/archives/C0FQSS30V/p1587754521457400) reading the search result with the announcement.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to understand the content of information, particularly as it relates to facility operating alerts.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: production

## Steps to Recreate

1. Enter this url in browser: https://www.va.gov/find-locations/?address=Washington%2C%20DC&context=Washington%2C%20District%20of%20Columbia%2C%20United%20States&facilityType=benefits&location=38.89317057287496%2C-77.03613281250001&page=1&serviceType&zoomLevel=9
2. Start screenreading device listed in Environment
3. Navigate to the search results, without looking at the screen
4. Verify that the alert context of the facility operating status is not conveyed

## Possible Fixes (optional)
**_Michelle's addendum_** _Let's provide context and indicate the temporary nature of this operating status without referencing COVID-19 so that the alerts remain flexible for changes in facility and service availability in the future (due to weather events, active shooter, etc)._ 

### Current code
```diff
<div class="usa-alert usa-alert-error background-color-only notice-marg-pad">
!  <i class="fa fa-exclamation-circle vads-u-margin-top--1 icon-base"></i>
  <div class="usa-alert-body">
    Facility Closed
  </div>
</div>
```

For each `<i>`, add attributes and screen reader only text:
- `aria-hidden="true"`
- `role="img"`
- <span class="sr-only">Alert: </span>
- <span class="sr-only"> temporarily due to covid-19</span>

### Recommended enhancement
```diff
<div class="usa-alert usa-alert-error background-color-only notice-marg-pad">
+  <i  aria-hidden="true" role="img" class="fa fa-exclamation-circle vads-u-margin-top--1 icon-base"></i>
+  <span class="sr-only">Alert: </span>
  <div class="usa-alert-body">
    Facility Closed
    <span class="sr-only"> temporarily due to covid-19</span>
  </div>
</div>
```

## WCAG or Vendor Guidance (optional)

* [W3C WCAG 2.0 - Semantically identifying a font icon with role="img"](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA24.html)

* [W3C Icon Font with an On-Screen Text Alternative](https://www.w3.org/WAI/GL/wiki/Icon_Font_with_an_On-Screen_Text_Alternative) - Preferably, according to W3C guidance, the alert label would appear on-screen: "The objective of this technique is to show how to provide a visible, text alternative for an icon font that conveys information. Icon fonts are fonts that use the Private Use Area (PUA) of Unicode. Typically they are inserted in HTML via the CSS @font-face declaration and generated content property. Since they are vectors they are scalable and resolution-independent. Icon fonts can have 2 problems:   
    - Some people with disabilities may not use assistive technology (AT) and rely on on-screen text alternatives.
    - For those who do use AT, voicing of icon fonts may be inaccurate, nonsensical, redundant, or unpredictable.
    - To solve these 2 problems aria-hidden="true" is used so AT will ignore the icon. Then an on-screen text alternative is added to convey meaning to everyone."

## Screenshots or Trace Logs

![Screen Shot 2020-04-23 at 2 09 31 PM](https://user-images.githubusercontent.com/57469/80133995-15631a80-856c-11ea-9a8c-7f1719845d3a.png)

![Screen Shot 2020-04-24 at 2 29 41 PM](https://user-images.githubusercontent.com/57469/80245480-c0410a80-8638-11ea-87ba-5d73c414341e.png)

**Comment by mmiddaugh** [comment]
@jenstrickland the team discussed this during grooming today. We'd like to say something more general like...._location has a temporary status of [operating status, like Closed]_. Would that provide the needed context? Any recommendations for implementation?

**Comment by jenstrickland** [comment]
@mmiddaugh Are y'all considering adding the additional context for all users, or only assistive tech?

**Comment by mmiddaugh** [comment]
After conversation with @jenstrickland, the following changes will make the alert acceptable:
Replace the unfriendly  icon description `class="fa fa-exclamation-circle vads-u-margin-top--1 icon-base"></i>` with <span class="sr-only">Alert: </span>. This will present alert context to Veterans with friendly concise language. 

Based on upcoming user research, we may add enhancements such as more meaningful titles or additional status options. Consider the editorial experience for this also. 



aria-hidden="true"
role="img"
Alert:
temporarily due to covid-19
Recommended enhancement
<div class="usa-alert usa-alert-error background-color-only notice-marg-pad">
+  <i  aria-hidden="true" role="img" class="fa fa-exclamation-circle vads-u-margin-top--1 icon-base"></i>
+  <span class="sr-only">Alert: </span>

**Comment by jenstrickland** [comment]
Here's sample code for the recommendation:

```html
<div class="usa-alert usa-alert-error background-color-only notice-marg-pad">
  <i class="fa fa-exclamation-circle vads-u-margin-top--1 icon-base" aria-hidden="true" role="presentation"></i>
  <span class="sr-only">Alert: </span>
  <div class="usa-alert-body">Facility Closed</div>
</div>
```

**Comment by mmiddaugh** [comment]
Ensure our handling of this is consistent with the method used on the Home page and the design system

**Comment by mpelzsherman** [comment]
Consider switching to design components: https://design.va.gov/components/alertboxes

**Comment by mmiddaugh** [comment]
@rtwell I reviewed the design system component page for alerts and did not see any reference to the `<span class="sr-only">Alert </span>` recommendation made by @jenstrickland in the thread above. 
If I missed it, my apologies. 
If her suggestion is not incorporated into the accessibility notes for the alert component, can we consider adding it? These alerts are appearing more frequently throughout VA.gov and we can create a more accessible and equitable experience by incorporating this into the design system. 
(I don't believe the big COVID banner such as on Pittsburgh) uses the span class approach either)

**Comment by mmiddaugh** [comment]
Just to close the loop above - design system council / team has the suggestion above in its backlog for prioritization

**Comment by mmiddaugh** [comment]
@jenstrickland this is ready for your validation when you are able. Thank you!

**Comment by jenstrickland** [comment]
Validated 🎉
